src/js/angular/import/controllers/import-view.controller.js

```diff
@@ -781,7 +781,8 @@ importViewModule.controller('UploadCtrl', ['$scope', 'toastr', '$controller', '$
     const updateTextImport = (settings) => {
         $scope.updating = true;
         ImportRestService.updateTextSnippet($repositories.getActiveRepository(), settings).success(function (data) {
-            // its updated
+            // Update list after file updates
+            $scope.updateListHttp(true);
```

Comment:
This is not enough to fix the issue. You can simulate the problem by slowing down your internet connection when loading the page, then try editing the query quickly.
IMO, we should return only the promise from the ImportRestService.updateTextSnippet call.
``` javascript
const updateTextImport = (settings) => {
        $scope.updating = true;
        return ImportRestService.updateTextSnippet($repositories.getActiveRepository(), settings).success(function (data) {
```
and where the updateTextImport function is called to be somethink like
``` javascript
updateTextImport()
  .then(() => $scope.updateListHttp(true))
  .then(() => {
  // import with updated model
});
```